pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -534,6 +534,44 @@ func (m *vmMutator) patchTerminationGracePeriodSeconds(vm *kubevirtv1.VirtualMac
 	return patchOps, nil
 }
 
+func patchCPUSocket(vm *kubevirtv1.VirtualMachine, patchOps types.PatchOps) []string {
+	cores := vm.Spec.Template.Spec.Domain.CPU.Cores
+	sockets := vm.Spec.Template.Spec.Domain.CPU.Sockets
+	threads := vm.Spec.Template.Spec.Domain.CPU.Threads
+
+	if cores == 0 {
+		cores = 1
+	}
+	if sockets == 0 {
+		sockets = 1
+	}
+	if threads == 0 {
+		threads = 1
+	}
```

Comment:
I think KubeVirt can already do this for us [here](https://github.com/kubevirt/kubevirt/blob/7a7199bbfa83b1f7585c607289644956cc186f66/pkg/defaults/defaults.go#L364-L387).

E.g., I run `k apply -f https://raw.githubusercontent.com/kubevirt/kubevirt/refs/heads/main/examples/vm-cirros.yaml` without setting any CPU topology in the VM, the VMI will be started with core, socket and thread set to 1.